src/bodywork/workflow_execution.py

```diff
@@ -400,3 +400,19 @@ def _remove_readonly(func: Any, path: Any, exc_info: Any) -> None:
         func(path)
     else:
         raise Exception
+
+
+def _ping_usage_stats_server() -> None:
+    """Pings the usage stats server."""
+    try:
+        param = "-n" if platform.system() == "Windows" else "-c"
+        output = run(
```

Comment:
I was thinking we'd do something like `requests.get(USAGE_STATS_SERVER_URL)`, so that the tracking server would receive a HTTP request, that is easy to handle with Flask/FastAPI.

Ping uses [Internet Control Message Protocol](https://en.wikipedia.org/wiki/Ping_(networking_utility)) - I have no idea how to work with this on the server-side (scratching at the limits of my knowledge here...).

What I originally had in mind, is repeating what we do for ` bodywork.workflow_execution.image_exists_on_dockerhub`, but without re-raising a `RunTimeError` if we get a `requests.exceptions.ConnectionError`.

What do you think?

Reply:
I misunderstood the ticket as a literal ping the server.

I don't think a normal http request is appropriate because we don't want to be waiting for the reply, if the server is unreachable it will cause a delay. If we set the timeout request to something short (to avoid waiting for a reply) like 1 second we'll end up throwing exceptions as part of normal process which isn't good.

I think the best option will be to use an async framework for an async request. I know this might seem like overkill for a ping but it's the best implementation if we're not going to use ping itself.